File: browserfn/src/main/java/org/sqlunet/browser/fn/DocumentTransformer.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.browser.fn;

import java.io.InputStream;

import androidx.annotation.Nullable;

/**
 * XSL Transformer
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class DocumentTransformer extends org.sqlunet.browser.web.DocumentTransformer
{
	static private final String XSL_DIR = "/org/sqlunet/";

	/**
	 * Get XSL file
	 *
	 * @param source     type of source
	 * @param isSelector is selector source
	 * @return XSL inputstream
	 */
	@Nullable
	protected InputStream getXSLStream(final String source, final boolean isSelector)
	{
		Settings.Source from = Settings.Source.valueOf(source);
		if (from == Settings.Source.FRAMENET)
		{
			final String xsl = DocumentTransformer.XSL_DIR + (isSelector ? "framenet2html-select.xsl" : "framenet2html.xsl");
			return DocumentTransformer.class.getResourceAsStream(xsl);
		}
		return null;
	}
}

File: browserfn/src/main/java/org/sqlunet/browser/fn/DocumentTransformer.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser.fn

import java.io.InputStream

/**
 * XSL Transformer
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
class DocumentTransformer : org.sqlunet.browser.web.DocumentTransformer() {

    /**
     * Get XSL file
     *
     * @param from       type of source
     * @param isSelector is selector source
     * @return XSL inputstream
     */
    override fun getXSLStream(from: String, isSelector: Boolean): InputStream? {
        val source = FnSettings.Source.valueOf(from)
        if (source == FnSettings.Source.FRAMENET) {
            val xsl = XSL_DIR + if (isSelector) "framenet2html-select.xsl" else "framenet2html.xsl"
            return DocumentTransformer::class.java.getResourceAsStream(xsl)
        }
        return null
    }

    companion object {
        private const val XSL_DIR = "/org/sqlunet/"
    }
}
